### PR TITLE
tests: Improve runtime of some tests when `--enable-debug`

### PR DESCRIPTION
**Pull request by achow101**
When configured with `--enable-debug`, many tests become dramatically slower. These slow downs are particularly noticed in tests that generate a lot of blocks in separate calls, make a lot of RPC calls, or send a lot of data from the test framework's P2P connection. This PR aims to improve the runtime of some of the slower tests and improve the overall runtime of the test runner. This has improved the runtime of the test runner from ~400s to ~140s on my computer.

The slowest test by far was `wallet_import_rescan.py`. This was taking ~320s. Most of that time was spent waiting for blocks to be mined and then synced to the other nodes. It was generating a new block for every new transaction it was creating in a setup loop. However it is not necessary to have one tx per block. By mining a block only every 10 txs, the runtime is improved to ~61s.

The second slowest test was `feature_fee_estimation.py`. This test spends most of its time waiting for RPCs to respond. I was able to improve its runtime by batching RPC requests. This has improved the runtime from ~201s to ~140s.

In `feature_taproot.py`, the test was constructing a Python `CScript` using a very large list of `OP_CHECKSIG`s. The constructor for the Python implementation of `CScript` was iterating this list in order to create a `bytes` from it even though a `bytes` could be created from it without iterating. By making the `bytes` before passing it into the constructor, we are able to improve this test's runtime from ~131s to ~106s.

Although `interface_rpc.py` was not typically a slow test, I found that it would occasionally have a super long runtime. It typically takes ~7s, but I have observed it taking >400s to run on occasion. This longer runtime occurs more often when `--enable-debug`. This long runtime was caused by the "exceeding work queue" test which is really just trying to trigger a race condition. In this test, it would create a few threads and try an RPC in a loop in the hopes that eventually one of the RPCs would be added to the work queue while another was processing. It used `getrpcinfo` for this, but this function is fairly fast. I believe what was happening was that with `--enable-debug`, all of the code for receiving the RPC would often take longer to run than the RPC itself, so the majority of the requests would succeed, until we got lucky after 10's of thousands of requests. By changing this to use a slow RPC, the race condition can be triggered more reliably, and much sooner as well. I've used `waitfornewblock` with a 500ms timeout. This improves the runtime to ~3s consistently.

The last test I've changed was `rpc_packages.py`. This test was one of the higher runtime variability tests. The main source of this variation appears to be waiting for the test node to relay a transaction to the test framework's P2P connection. By whitelisting that peer, the variability is reduced to nearly 0.

Lastly, I've reordered the tests in `test_runner.py` to account for the slower runtimes when configured with `--enable-debug`. Some of the slow tests I've looked at were listed as being fast which was causing overall `test_runner.py` runtime to be extended. This change makes the test runner's runtime be bounded by the slowest test (currently `feature_fee_estimation.py` with my usual config (`-j 60`).

### CODE REVIEW

File: test/functional/wallet_import_rescan.py [hunk 1]
```diff
@@ -217,11 +232,14 @@ def run_test(self):
...
-        for variant in IMPORT_VARIANTS:
+        for i, variant in enumerate(IMPORT_VARIANTS):
+            if i % 10 == 0:
+                blockhash = self.generate(self.nodes[0], 1)[0]
+                conf_height = self.nodes[0].getblockcount() + 1
             variant.sent_amount = get_rand_amount()
             variant.sent_txid = self.nodes[0].sendtoaddress(variant.address["address"], variant.sent_amount)
-            self.generate(self.nodes[0], 1)  # Generate one block for each send
-            variant.confirmation_height = self.nodes[0].getblockcount()
+            variant.confirmation_height = conf_height;
```
Comment (MarcoFalke) [inline test/functional/wallet_import_rescan.py:241]:
```
test/functional/wallet_import_rescan.py:241:54: E703 statement ends with a semicolon

File: test/functional/rpc_packages.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ class RPCPackagesTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
+        self.extra_args = [["-whitelist=127.0.0.1"]]
```
Comment (MarcoFalke) [inline test/functional/rpc_packages.py:32]: nit: Could only add the needed permission flag? and add a comment?

**Review by maflcko — APPROVED** [review]
lgtm

**Comment by fanquake** [comment]
Concept ACK - seeing significant speedup when running the modified tests under `--enable-debug`. `test/functional/test_runner.py wallet_import_rescan.py` drops from 418s to 106s.

File: test/functional/wallet_import_rescan.py (hunk 1 above)
Comment (achow101) [reply test/functional/wallet_import_rescan.py]: Fixed

File: test/functional/rpc_packages.py (hunk 2 above)
Comment (achow101) [reply test/functional/rpc_packages.py]: Done

**Comment by willcl-ark** [comment]
Concept ACK!

This is great to see and is something that has been bothering me for quite some time.

Without this PR tests for me were taking 268 seconds to run, where:

- `wallet_import_rescan` 227s, 
- `feature_taproot` 179s,
- `feature_fee_estimation` 267s and 
- `feature-block` 167s.

After this PR my total test time reduced to 171 seconds, where:

- `wallet_import_rescan` 49s!, 
- `feature_taproot` 148s, 
- `feature_fee_estimation` 164s and 
- `feature_block` 170s.

This leaves me with the following tests taking over 1 minute runtime:

```
❯ cat times | sort -k 3 -n -t "|" -r
feature_block.py                                       | ✓ Passed  | 170 s
feature_fee_estimation.py                              | ✓ Passed  | 164 s
feature_taproot.py                                     | ✓ Passed  | 148 s
mempool_updatefromblock.py                             | ✓ Passed  | 112 s
p2p_segwit.py                                          | ✓ Passed  | 107 s
p2p_blockfilters.py                                    | ✓ Passed  | 100 s
feature_assumevalid.py                                 | ✓ Passed  | 99 s
wallet_taproot.py --descriptors                        | ✓ Passed  | 92 s
p2p_headers_sync_with_minchainwork.py                  | ✓ Passed  | 92 s
wallet_avoidreuse.py --legacy-wallet                   | ✓ Passed  | 87 s
p2p_invalid_messages.py                                | ✓ Passed  | 85 s
wallet_groups.py --descriptors                         | ✓ Passed  | 76 s
wallet_avoidreuse.py --descriptors                     | ✓ Passed  | 76 s
feature_maxuploadtarget.py                             | ✓ Passed  | 71 s
mining_getblocktemplate_longpoll.py                    | ✓ Passed  | 69 s
wallet_groups.py --legacy-wallet                       | ✓ Passed  | 63 s
wallet_fundrawtransaction.py --descriptors             | ✓ Passed  | 60 s
```

**Review by willcl-ark — APPROVED** [review]
ACK 1647a11

The final tests to finish for me with `--enable-debug` and 50 jobs are generally:

```
245/255 - feature_coinstatsindex.py passed, Duration: 25 s
246/255 - p2p_headers_sync_with_minchainwork.py passed, Duration: 93 s
247/255 - wallet_taproot.py --descriptors passed, Duration: 94 s
248/255 - feature_config_args.py passed, Duration: 35 s
249/255 - p2p_blockfilters.py passed, Duration: 105 s
250/255 - feature_assumevalid.py passed, Duration: 105 s
251/255 - p2p_segwit.py passed, Duration: 107 s
252/255 - mempool_updatefromblock.py passed, Duration: 116 s
253/255 - feature_taproot.py passed, Duration: 155 s
254/255 - feature_block.py passed, Duration: 175 s
255/255 - feature_fee_estimation.py passed, Duration: 209 s
```

So the new ordering of the list feels pretty good.

**Pull request merged**